examples/templates/scratch/main.tf

```
@@ -0,0 +1,64 @@
+terraform {
+  required_providers {
+    coder = {
+      source = "coder/coder"
+    }
+  }
+}
+
+data "coder_provisioner" "me" {}
+
+data "coder_workspace" "me" {}
+
+resource "coder_agent" "main" {
+  arch                   = data.coder_provisioner.me.arch
+  os                     = data.coder_provisioner.me.os
+  startup_script_timeout = 180
+  startup_script         = <<-EOT
+    set -e
+    # Run programs at workspace startup
+  EOT
+
+  metadata {
+    display_name = "CPU Usage"
+    key          = "0_cpu_usage"
+    script       = "coder stat cpu"
+    interval     = 10
+    timeout      = 1
+  }
+
+  metadata {
+    display_name = "RAM Usage"
+    key          = "1_ram_usage"
+    script       = "coder stat mem"
+    interval     = 10
+    timeout      = 1
+  }
+}
+
+# Use this to set environment variables in your workspace
+# details: https://registry.terraform.io/providers/coder/coder/latest/docs/resources/env
+resource "coder_env" "my_env" {
+  agent_id = coder_agent.main.id
+  name     = "FOO"
+  value    = "bar"
+}
```

Perhaps this could be something more inviting?

```suggestion
resource "coder_env" "welcome_message" {
  agent_id = coder_agent.main.id
  name     = "WELCOME_MESSAGE"
  value    = "Welcome to your Coder workspace!"
```